src/codelens.ts

```diff
@@ -0,0 +1,125 @@
+import * as vscode from 'vscode';
+import * as statusbar from './statusbar';
+import { juvixRoot } from './root';
+import { isJuvixFile } from './utils/base';
+/**
+ * CodelensProvider
+ */
+
+export function activate(context: vscode.ExtensionContext) {
+    context.subscriptions.push(
+        vscode.languages.registerCodeLensProvider(
+            { scheme: 'file', language: 'Juvix' },
+            new CodelensProvider()
+        )
+    );
+    context.subscriptions.push(
+        vscode.commands.registerCommand("juvix-mode.enableCodeLens", () => {
+            vscode.workspace.getConfiguration("juvix-mode").update("enableCodeLens", true, true);
+        })
+    );
+    context.subscriptions.push(
+        vscode.commands.registerCommand("juvix-mode.disableCodeLens", () => {
+            vscode.workspace.getConfiguration("juvix-mode").update("enableCodeLens", false, true);
+        })
+    );
+    context.subscriptions.push(
+        vscode.commands.registerCommand("juvix-mode.codelensAction", (args: any) => {
+            vscode.window.showInformationMessage(`CodeLens action clicked with args=${args}`);
+        })
+    );
+
+    context.subscriptions.push(
+        vscode.commands.registerCommand("juvix-mode.aux.prependText", (args: any) => {
+            const editor = vscode.window.activeTextEditor;
+            if (editor) {
+                const position = new vscode.Position(0, 0);
+                editor.edit((editBuilder) => {
+                    editBuilder.insert(position, args.text);
+                });
+            }
+        })
+    );
+}
+
+export class CodelensProvider implements vscode.CodeLensProvider {
+
+    private codeLenses: vscode.CodeLens[] = [];
+    private _onDidChangeCodeLenses: vscode.EventEmitter<void> = new vscode.EventEmitter<void>();
+    public readonly onDidChangeCodeLenses: vscode.Event<void> = this._onDidChangeCodeLenses.event;
+
+    constructor() {
+        vscode.workspace.onDidChangeConfiguration((_) => {
+            this._onDidChangeCodeLenses.fire();
+        });
+    }
+
+    public provideCodeLenses(
+        document: vscode.TextDocument
+        , _token: vscode.CancellationToken)
+        : vscode.CodeLens[] | Thenable<vscode.CodeLens[]> {
+
+        if (vscode.workspace
+            .getConfiguration("JuvixCodeLens")
+            .get("enableCodeLens", true)) {
+
+            this.codeLenses = [];
+            const text = document.getText();
+
+            /*
+                Add a code lenses that checks if the document is empty.
+                If it is, it suggests to insert the module header
+                relative to juvixRoot.
+                The content inserted by the code lenses should be:
+                   "module <module name>;"
```

Comment:
For some reason, in the test folder of the Juvix repo, when I tried it, it added the whole path (even though, maybe there should not be roots, as it is not the juvix project?):

<img width="718" alt="Screenshot 2023-05-26 at 17 25 40" src="https://github.com/anoma/vscode-juvix/assets/8126674/0eca2157-46cf-4791-af8f-a37f54ddac7e">

Reply:
Try it with the last commit. It should be fixed.

Reply:
I don't get the suggestion for the name anywhere anymore. Should there be any condition so that it will suggest the name?
Can not get it work even on the example from the description of this PR 🤔

Reply:
Still get the same in the juvix test folder:

<img width="586" alt="Screenshot 2023-05-30 at 12 12 09" src="https://github.com/anoma/vscode-juvix/assets/8126674/e0863b95-aea4-445a-a646-c778a21c34a5">

But `vscode-juvix` `HelloWorld` example now autoadds the module name